### PR TITLE
Add Xpanse data and update Vulnerability Details page with sections based on source or CVE

### LINKED ISSUE #330: UI: Create Vulnerability Details visuals for Xpanse data
# 💡 Summary #

Create Vulnerability Details visuals for Xpanse data 

## Motivation and context ##

Different vulnerability details will show different content. This is specific to Xpanse data.
JIRA Ticket #: CRASM-338

## Implementation notes ##

The vulnerability table has a field called "source". When it has the field "xpanse", it will show a different set of details in the UI.

## Acceptance criteria ##

- [ ] When clicking on details in the vulnerability table, it should show this updated page for an xpanse vulnerability.

### PR DESCRIPTION
**Pull request by ameliav**
#  #

## 🗣 Description ##
Add Xpanse data and update Vulnerability Details page with sections based on source or CVE.

## 💭 Motivation and context ##
Added a new data source from Xpanse which does not usually include CVEs.
The Vulnerability Details page will now be visually different based on:
1. If it is source field is Palo Alto Xpanse
2. If it has a proper CVE in the CVE field
3. If it does not have a CVE

The fields that are null should also now not break the page and be shown as N/A.

JIRA Ticket #: CRASM-338
Screenshots are shown in JIRA.

## 🧪 Testing ##

Going to any vulnerability details page should show you one of three different displays.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: backend/src/tasks/xpanse-sync.ts [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+  }
+};
+const getVulnData = async (
+  org: Organization,
+  commandOptions: CommandOptions
+) => {
+  let done = false;
+  let page = 1;
+  let total_pages = 1;
+  // let fullVulnArray: XpanseVulnOutput[] = [];
```
Comment (cduhn17) [inline backend/src/tasks/xpanse-sync.ts]:
```suggestion

```
*Multipe locations. Remove commented out lines or add TODO and issues that detail feature expansion

**Review by cduhn17 — COMMENTED** [review]
See comments

File: backend/src/tasks/xpanse-sync.ts (hunk 1 above)
Comment (DJensen94) [reply backend/src/tasks/xpanse-sync.ts]: Addressed in the following commit: [ea2524d](https://github.com/cisagov/XFD/pull/346/commits/ea2524d852037e246e65942a1c48a63961fa9287)

**Review by schmelz21 — APPROVED** [review]
LGTM with updated commits from - https://github.com/cisagov/XFD/pull/346#discussion_r1644836133

**Review by cduhn17 — APPROVED** [review]
LGTM

**Pull request merged**